Base Packages and Essentials Survey (July, 2024)

MingcongBai opened the pull request:
Topic Description
-----------------

- zstd: bump REL for topic Revision Marking Guidelines
- iw: bump REL for topic Revision Marking Guidelines
- dhcp: bump REL for topic Revision Marking Guidelines
- wpa-supplicant: bump REL for topic Revision Marking Guidelines
- aosc-aaa: bump REL for topic Revision Marking Guidelines
- zlib: bump REL for topic Revision Marking Guidelines
- gcc: bump REL for topic Revision Marking Guidelines
- wget: bump REL for topic Revision Marking Guidelines
- curl: bump REL for topic Revision Marking Guidelines
- xz: bump REL for topic Revision Marking Guidelines

... and 129 more commits

Package(s) Affected
-------------------

- admin-base: 14-1
- aosc-aaa: 11.5.0-2
- apt: 2:2.6.1-4
- bash: 5.2.26-2
- benchmark-base: 1-1
- boot-base: 5-1
- cinnamon-base: 2-1
- codec-base: 2-1
- coreutils: 9.5-2
- curl: 8.9.1-2
- dbus: 2:1.14.10-3
- debug-base: 3-1
- desktop-base: 12-1
- dhcp: 4.4.3.P1-2
- diffutils: 3.10-2
- dpkg: 1.22.7-2
- fcitx-base: 1-1
- fcitx5-base: 4-1
- findutils: 4.10.0-2
- gawk: 5.3.0-2
- gcc: 13.2.0-8
- gcc-runtime: 13.2.0-8
- gnome-base: 5-1
- grep: 3.11-2
- gzip: 1:1.13-2
- ibus-base: 1-1
- iproute2: 6.10.0-2
- iw: 5.19-2
- kbd: 2.5.1-2
- kernel-base: 4-1
- kmod: 31-2
- less: 1:643-3
- lxde-base: 2-1
- mate-base: 4-1
- nemo: 5.8.3-4
- net-tools: 1:2.10-3
- network-base: 6-1
- networkmanager: 1.40.6-3
- oma: 1.4.1-2
- print-base: 1-1
- procps: 4.0.4-2
- psmisc: 23.6-2
- python-base: 2-1
- retro-server-base: 1-1
- sed: 4.9-2
- server-base: 4-1
- shadow: 4.15.1-2
- sudo: 1.9.15p5-2
- tar: 1.35-2
- update-alternatives: 1.22.7-2
- util-base: 12-1
- util-linux: 2.40.1-2
- util-linux-runtime: 2.40.1-2
- web-base: 1-1
- wget: 1.24.5-2
- which: 2.21-5
- wpa-supplicant: 1:2.11-2
- xfce-base: 4-1
- xz: 5.6.2-2
- zlib: 1.3.1-3
- zstd: 1.5.6-2

Security Update?
----------------

No

Build Order
-----------

```
#buildit admin-base benchmark-base boot-base nemo cinnamon-base codec-base debug-base desktop-base fcitx5-base fcitx-base gnome-base ibus-base kernel-base lxde-base mate-base network-base print-base python-base retro-server-base server-base util-base web-base xfce-base apt dbus kmod oma shadow sudo iproute2 net-tools networkmanager bash coreutils diffutils findutils gawk grep gzip kbd less procps psmisc sed tar util-linux which xz curl wget gcc zlib aosc-aaa wpa-supplicant dhcp iw zstd
```

For architectures without GNOME (loongarch64, loongson3, ppc64el):

```
/build base-package-rework-survey-20240716 autobuild4,admin-base,benchmark-base,boot-base,nemo,cinnamon-base,codec-base,debug-base,desktop-base,fcitx5-base,fcitx-base,ibus-base,kernel-base,lxde-base,mate-base,network-base,print-base,python-base,retro-server-base,server-base,util-base,web-base,xfce-base,apt,dbus,kmod,oma,shadow,sudo,iproute2,net-tools,networkmanager,bash,coreutils,diffutils,findutils,gawk,grep,gzip,kbd,less,procps,psmisc,sed,tar,util-linux,which,xz,curl,wget,gcc,zlib,aosc-aaa,wpa-supplicant,dhcp,iw,zstd loongarch64,ppc64el,loongson3
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

eatradish committed:
util-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
admin-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
boot-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
codec-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
fcitx-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
cinnamon-base: move non-essential components to PKGRECOM

MingcongBai committed:
astronomy-base: drop, orphaned

MingcongBai committed:
benchmark-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
compiz-base: drop, orphaned

MingcongBai committed:
container-base: drop, orphaned

This is so pointless it hurts.

MingcongBai committed:
debug-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
desktop-base: drop unused BUILDDEP

MingcongBai committed:
fcitx5-base: drop unused BUILDDEP

MingcongBai committed:
gnome-base: move non-essential components to PKGRECOM

MingcongBai committed:
gtk-base: drop, orphaned

MingcongBai committed:
hometheatre-base: drop, orphaned

MingcongBai committed:
i3-base: drop, orphaned

MingcongBai committed:
ibus-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
interdevice-base: drop, orphaned

MingcongBai committed:
kernel-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
localization-base: drop, orphaned

MingcongBai committed:
lxde-base: move non-essential components to PKGRECOM

MingcongBai committed:
creativity-base: drop, orphaned

MingcongBai committed:
mate-base: move non-essential components to PKGRECOM

MingcongBai committed:
network-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
openbox-base: drop, orphaned

MingcongBai committed:
photography-base: drop, orphaned

MingcongBai committed:
print-base: move non-essential components to PKGRECOM

MingcongBai committed:
python-base: move non-essential components to PKGRECOM

MingcongBai committed:
qt-base: drop, orphaned

MingcongBai committed:
retro-server-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
ruby-base: drop, orphaned

MingcongBai committed:
rust-base: drop, orphaned

MingcongBai committed:
scan-base: drop, orphaned

MingcongBai committed:
sniffing-base: drop, orphaned

MingcongBai committed:
studio-base: drop, orphaned

MingcongBai committed:
server-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
tex-base: drop, orphaned

MingcongBai committed:
virtualization-base: drop, orphaned

MingcongBai committed:
way-cooler-base: drop, orphaned

MingcongBai committed:
web-base: move all PKGDEP to PKGRECOM

MingcongBai committed:
wine-base: drop, orphaned

MingcongBai committed:
wx-base: drop, orphaned

MingcongBai committed:
xfce-base: move non-essential components to PKGRECOM

MingcongBai committed:
oma: mark as Essential

MingcongBai committed:
apt: mark as Essential

Sync PKGEPOCH for Spiral.

FIXME: Autobuild should exclude self from Spiral provides.

MingcongBai committed:
dpkg: mark as Essential

MingcongBai committed:
dbus: mark as Essential

MingcongBai committed:
kmod: mark as Essential

MingcongBai committed:
diffutils: mark as Essential

MingcongBai committed:
bash: mark as Essential

MingcongBai committed:
sudo: mark as Essential

MingcongBai committed:
shadow: mark as Essential

MingcongBai committed:
iproute2: mark as Essential

MingcongBai committed:
net-tools: mark as Essential

MingcongBai committed:
findutils: mark as Essential

MingcongBai committed:
coreutils: mark as Essential

MingcongBai committed:
networkmanager: mark as Essential

MingcongBai committed:
kbd: mark as Essential

MingcongBai committed:
less: mark as Essential

MingcongBai committed:
util-linux: mark as Essential

MingcongBai committed:
procps: mark as Essential

MingcongBai committed:
gawk: mark as Essential

MingcongBai committed:
psmisc: mark as Essential

MingcongBai committed:
sed: mark as Essential

MingcongBai committed:
gzip: mark as Essential

MingcongBai committed:
tar: mark as Essential

MingcongBai committed:
grep: mark as Essential

MingcongBai committed:
gcc-runtime: mark as Essential

MingcongBai committed:
aosc-aaa: mark as Essential

MingcongBai committed:
wget: mark as Essential

MingcongBai committed:
which: mark as Essential

MingcongBai committed:
xz: mark as Essential

MingcongBai committed:
curl: mark as Essential

MingcongBai committed:
zlib: mark as Essential

cthbleachbit commented:
Stuff that probably shouldn't be ESSENTIAL:
- `inetutils`: curl can be used as an ftp client, `hostname` command is replaced by `hostnamectl` from systemd.
- `net-tools`: superseded by iproute2 in terms of function
- `patch`: probably won't be game-over if uninstalled

Stuff that probably should be ESSENTIAL:
- `wpa-supplicant` / `dhclient` / `iw`: core connectivity utilities used by NM or sysadmin manually
- `zstd` - AOSC OS ships zstd compressed kernel modules and firmware blobs. system might not survive after a reboot without zstd.

KexyBiscuit commented:
Merging #7230 into this. Loop @MingcongBai

MingcongBai committed:
wpa-supplicant: mark as Essential

MingcongBai committed:
dhcp: mark as Essential

MingcongBai committed:
iw: mark as Essential

MingcongBai committed:
zstd: mark as Essential

MingcongBai commented:
> Stuff that probably shouldn't be ESSENTIAL:
> 
>     * `inetutils`: curl can be used as an ftp client, `hostname` command is replaced by `hostnamectl` from systemd.
> 
>     * `net-tools`: superseded by iproute2 in terms of function
> 
>     * `patch`: probably won't be game-over if uninstalled
> 
> 
> Stuff that probably should be ESSENTIAL:
> 
>     * `wpa-supplicant` / `dhclient` / `iw`: core connectivity utilities used by NM or sysadmin manually
> 
>     * `zstd` - AOSC OS ships zstd compressed kernel modules and firmware blobs. system might not survive after a reboot without zstd.

Addressed with the exception of `net-tools`. I for one still prefers `ifconfig`.

MingcongBai committed:
nemo: disable tracker support

FIXME: Re-enable Tracker support when we update to a version that supports
Tracker 3.

KatyushaScarlet commented:
Stuff that probably shouldn't be ESSENTIAL in `server-base`:
* `lnmp-base` / `lamp-base` : For web servers, there are various environmental options available, such as Nginx/Apache/Caddy + MariaDB/PostgreSQL/... + (optional) PHP5/PHP7/PHP8 + .... Typically, a web server will select a specific combination that best suits its web application, compiling or upgrading the environment independently. Moreover, changing the version of software packages like PHP (upgrading from 5 to 7, or from 7 to 8) may break compatibility with the web application. I suggest not including `lnmp-base` / `lamp-base` in `server-base` because users' preferences vary, and choosing a default environment does not have universal applicability.

KexyBiscuit commented on meta-bases/benchmark-base/spec:
```diff
@@ -1,3 +1,2 @@
-VER=0
-REL=1
+VER=2
```
2 or 1?

KexyBiscuit commented on meta-bases/python-base/autobuild/defines:
```diff
@@ -1,6 +1,7 @@
 PKGNAME=python-base
 PKGSEC=Bases
-PKGDEP="python-2 python-3 pip virtualenv"
+PKGDEP="python-2 python-3"
```
Python 2 until the finale.

KexyBiscuit reviewed (dismissed):
LGTM overall.

MingcongBai committed:
nemo: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
codec-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
cinnamon-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
benchmark-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
admin-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
boot-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
print-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
network-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
mate-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
fcitx5-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
lxde-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
ibus-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
server-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
util-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
debug-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
xfce-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
gnome-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
fcitx-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
apt: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
web-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
retro-server-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
dbus: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
python-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
kernel-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
desktop-base: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
grep: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
coreutils: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
sudo: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
gawk: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
gzip: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
net-tools: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
oma: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
iproute2: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
diffutils: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
networkmanager: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
findutils: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
shadow: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
dpkg: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
bash: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
less: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
kbd: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
procps: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
kmod: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
zstd: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
gcc: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
zlib: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
iw: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
dhcp: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
curl: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
aosc-aaa: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
which: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
util-linux: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
xz: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
tar: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
psmisc: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
wget: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
wpa-supplicant: bump REL for topic Revision Marking Guidelines

MingcongBai committed:
sed: bump REL for topic Revision Marking Guidelines
